### PR TITLE
test: add unit tests for echo response parser

### PR DESCRIPTION
**Pull request by archy-rock3t-cloud**
**What type of PR is this?**

/kind test
/area conformance-machinery

**What this PR does / why we need it**:

The `conformance/utils/echo` package has no test coverage. This adds unit tests for `ParseResponse`, `parseMultipleResponses`, `GetHeaders`, and `Body` — covering field extraction from echo server output, case-insensitive request ID matching, header accumulation semantics (Add vs Set), body key-value parsing, multi-response splitting, and edge cases like missing fields and malformed header lines.

**Which issue(s) this PR fixes**:

N/A — test coverage improvement

**Does this PR introduce a user-facihange?**:

```release-note
NONE
```

### CODE REVIEW

File: conformance/utils/echo/parse_test.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+				assert.Equal(t, "example.com", r.Host)
+				assert.Equal(t, "echo-pod-abc", r.Hostname)
+				assert.Equal(t, "/status/200", r.URL)
+				assert.Equal(t, "kind-cluster", r.Cluster)
+				assert.Equal(t, "10.244.0.5", r.IP)
+			},
+		},
+		{
+			name:   "request ID match is case-insensitive",
+			output: "x-request-id=lowercase-id",
```
Comment (rikatz) [inline conformance/utils/echo/parse_test.go:77]: if you are testing case insensitive, shouldn't you make this output something like CamelCase and assert that r.ID is all lower?

**Comment by rikatz** [comment]
the PR overall lgtm, I have a minor comment but it is not a blocker and can be followed up

**Comment by rikatz** [comment]
/approve
/lgtm
thanks!
